helper-cli/src/main/kotlin/commands/GroupScanIssuesCommand.kt

```diff
@@ -0,0 +1,113 @@
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.utils.common.expandTilde
+
+class GroupScanIssuesCommand : CliktCommand(
+    help = "Shows the amount of affected dependencies for each scan issue category."
+) {
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val ortResult = ortFile.readValue<OrtResult>()
+
+        val issues = ortResult.getScannerIssues(
+            omitExcluded = true,
+            omitResolved = true,
+            minSeverity = Severity.ERROR
+        ).filter { (id, _) -> ortResult.isPackage(id) }
+
+        val issueCategoriesForId = issues.mapValues { issue ->
+            issue.value.mapTo(mutableSetOf()) { it.category }
+        }
+
+        val issueCategoriesForIdWithoutVersion = issueCategoriesForId.entries.groupBy(
+            { it.key.copy(version = "") },
+            { it.value }
+        ).mapValues { it.value.flatten().toSet() }
+
+        val pkgCountsForIssueCategory = ScanIssueCategory.entries.associateWith { category ->
+            val numPackages = issueCategoriesForId.count { (_, categories) ->
+                category in categories
+            }
+
+            val numPackagesWithoutVersion = issueCategoriesForIdWithoutVersion.count { (_, categories) ->
+                category in categories
+            }
+
+            numPackages to numPackagesWithoutVersion
+        }
+
+        val stats = buildString {
+            pkgCountsForIssueCategory.entries.sortedByDescending { it.value.first }.forEach { (category, counts) ->
+                appendLine("$category: ${counts.first} / ${counts.second}")
+            }
+        }
+
+        print(stats)
+    }
+}
+
+private enum class ScanIssueCategory(
+    val regex: Regex
+) {
+    PROVENANCE_INFO_MISSING(
+        "IOException: Could not resolve provenance for package '.*' for source code origins \\[.*\\]\\."
+    ),
+    SCAN_TIMED_OUT(
+        "ERROR: Timeout after .* seconds while scanning file '.*'\\."
+    ),
+    VCS_REVISION_NOT_FOUND(
+        ".*Could not resolve revision.*Could not find any revision candidates.*"
+    ),
+    VCS_PATH_NOT_EXISTENT(
+        ".*Could not resolve provenance.*because the requested VCS path.*does not exist."
+    ),
+    VCS_TYPE_UNKNOWN(
+        ".*Could not determine VCS for type.*"
+    ),
+    OTHER("");
+
+    constructor(regex: String) : this(regex.toRegex(setOf(RegexOption.DOT_MATCHES_ALL, RegexOption.MULTILINE)))
```

Comment:
Nit: At this point `regex` is not a `Regex` instance yet, but a `String` pattern _for_ a `Regex`, so maybe call the variable "pattern" instead?